File: c++/temperatura.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {

    double c, f;
    char escala;

    cout << "Voce vai digitar a temperatura em qual escala (C/F)? ";
    cin >> escala;

    if (escala == 'f') {
        cout << "Digite a temperatura em Fahrenheit: ";
        cin >> f;
        c = (5 * ( f - 32)) / 9;
        cout << fixed << setprecision(2);
        cout << "Temperatura equivalente em Celsius: " << c << endl;
    } else {
        cout << "Digite a temperatura em Celsius: ";
        cin >> c;
        f = (c * 9 / 5) + 32;
        cout << fixed << setprecision(2);
        cout << "Temperatura equivalente em Fahrenheit: " << f << endl;
    }

    return 0;
}

File: java/temperatura.java
import java.util.Locale;
import java.util.Scanner;

public class temperatura {

	public static void main(String[] args) {
		
		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		
		double c, f;
		char temperatura;
		
		System.out.print("Voce vai digitar a temperatura em qual escala (C/F)? ");
		temperatura = sc.next().charAt(0);
		
		if (temperatura == 'f') {
			System.out.print("Digite a temperatura em Fahrenheit: ");
			f = sc.nextDouble();
			c = 5 * (f - 32) / 9;
			System.out.println("Temperatura equivalente em Celsius: " + String.format("%.2f", c));
		} else {
			System.out.print("Digite a temperatura em Celsius: ");
			c = sc.nextDouble();
			f = (9 * c / 5) + 32;
			System.out.println("Temperatura equivalente em Fahrenheit: " + String.format("%.2f", f));
		}
		
		sc.close();
	}

}
